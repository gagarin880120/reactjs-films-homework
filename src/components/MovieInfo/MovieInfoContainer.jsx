import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MovieInfo from './MovieInfo';
import { movieDetailsSelector } from '../../redux/selectors';

export function MovieInfoContainer({ movie }) {
  return (
    <MovieInfo
      movie={movie}
    />
  );
}

export const mapStateToProps = (state) => ({
  movie: movieDetailsSelector(state),
});

MovieInfoContainer.propTypes = {
  movie: PropTypes.instanceOf(Object),
};

MovieInfoContainer.defaultProps = {
  movie: {},
};

export default connect(mapStateToProps)(MovieInfoContainer);
