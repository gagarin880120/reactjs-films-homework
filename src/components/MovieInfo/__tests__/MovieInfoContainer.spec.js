import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import { MovieInfoContainer, mapStateToProps } from '../MovieInfoContainer';

describe('MovieInfoContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<MovieInfoContainer />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  })

  describe('mapStateToProps', () => {
    test('should return the right value', () => {

      const initialState = {
        movieDetails: {title: 'Matrix', runtime: 120, vote_average: 8.6, genres: [{name: 'Adventure'}]},
      };

      expect(mapStateToProps(initialState).movie).toEqual({
        genres: 'Adventure',
        runtime: '2h',
        stars: [
          {
            id: 0,
            type: 'full',
          },
          {
            id: 1,
            type: 'full',
          },
          {
            id: 2,
            type: 'full',
          },
          {
            id: 3,
            type: 'full',
          },
          {
            id: 4,
            type: 'half',
          },
        ],
        title: 'Matrix',
        vote_average: 8.6,
      });
    });
  })
})
