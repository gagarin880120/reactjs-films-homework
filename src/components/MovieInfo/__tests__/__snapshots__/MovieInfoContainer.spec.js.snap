// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoContainer component should render without crashing 1`] = `
<MovieInfo
  movie={Object {}}
/>
`;
